### ISSUE TITLE
Error while saving YAML

### ISSUE THREAD

**Issue by VenkateshSrini** [kubernetes-client/csharp#554]
Hi, 
The below is the code that I have written clone the Kubernetes job. I have getting an error "UnProcessibleEntity". So I tried to serialize yaml to see what was issue. But I get a vague error. 

The code is as below

    
```
[Route("api/[controller]")]
    [ApiController]
    public class JobsController : ControllerBase
    {
        private readonly IKubernetesClientFactory _k8sFactory;
        readonly KubernetesClusterConfigData _k8sClusterConnectionConfig;
        readonly ILogger<JobsController> _logger;
        public JobsController(IKubernetesClientFactory k8sFactory,
            IOptions<KubernetesClusterConfigData> k8sConnectionConfigOptions, ILogger<JobsController> logger)
        {
            _k8sFactory = k8sFactory;
            _k8sClusterConnectionConfig = k8sConnectionConfigOptions.Value;
            _logger = logger;

        }
        /// <summary>
        /// Creates a clone from existing  job
        /// </summary>
        /// <param name="request"> job request</param>
        /// <returns>an instance of ResCreateFromCronJob n HTTPMessage</returns>
        [HttpPost]
        [Route("CreateJobFromExisting")]
        public async Task<ActionResult<ResCreateFromCronJob>> CreateJobFromExisting(ReqCreateFromJob request)
        {
            var k8sClient = _k8sFactory.K8sClient;
            V1Container container;
            var existingJob = await k8sClient.ReadNamespacedJobAsync(request?.ExistigJobName, request.NameSpace);
            if (!string.IsNullOrEmpty(request?.ContainerName))
                container = existingJob.Spec.Template.Spec.Containers?
                            .FirstOrDefault(cntair => cntair.Name.CompareTo(request?.ContainerName) == 0);
            else
                container = existingJob.Spec.Template.Spec.Containers[0];

            if (request.Commands.Any())
                container.Command = request.Commands;
            if (request.JobParameters.Any())
                container.Args = request.JobParameters;
            
            var newJob = new V1Job
            {
                ApiVersion = existingJob.ApiVersion,
                Kind = existingJob.Kind,
                Metadata = existingJob.Metadata,
                Spec = existingJob.Spec
            };
            newJob.Metadata.Name = request?.NewJobName;
            newJob.Metadata.NamespaceProperty = request?.NameSpace;
            await WriteYaml(newJob);
            var newJobCreationResult = await k8sClient.CreateNamespacedJobAsync(newJob, request?.NameSpace);
            var response = new ResCreateFromCronJob
            {
                JobName = newJobCreationResult.Name(),
                JobUid = newJobCreationResult.Uid(),
                NameSpace = newJobCreationResult.Namespace()
            };
            _logger.LogInformation("New job created scuccessfully");
            return Ok(response);
        }
        private async Task WriteYaml(V1Job job)
        {
            var yamlString = Yaml.SaveToString<V1Job>(job);
            await System.IO.File.WriteAllTextAsync("newjob.yaml", yamlString);
        }
    }

```

The exception that I'm getting is 

> An unhandled exception has occurred while executing the request.
System.Reflection.TargetException: Object does not match target type.
   at System.Reflection.RuntimeMethodInfo.CheckConsistency(Object target)
   at System.Reflection.RuntimeMethodInfo.InvokeArgumentsCheck(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at YamlDotNet.Helpers.GenericDictionaryToNonGenericAdapter.DictionaryEnumerator.get_Key()
   at YamlDotNet.Helpers.GenericDictionaryToNonGenericAdapter.DictionaryEnumerator.get_Entry()
   at YamlDotNet.Helpers.GenericDictionaryToNonGenericAdapter.DictionaryEnumerator.get_Current()
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseDictionary[TContext](IObjectDescriptor dictionary, IObjectGraphVisitor`1 visitor, Type keyType, Type valueType, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseObject[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.Traverse[TContext](Object name, IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseProperties[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseObject[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.Traverse[TContext](Object name, IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseList[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseObject[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.Traverse[TContext](Object name, IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseProperties[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseObject[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.Traverse[TContext](Object name, IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseProperties[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseObject[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.Traverse[TContext](Object name, IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
   at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.YamlDotNet.Serialization.IObjectGraphTraversalStrategy.Traverse[TContext](IObjectDescriptor graph, IObjectGraphVisitor`1 visitor, TContext context)
   at YamlDotNet.Serialization.SerializerBuilder.ValueSerializer.SerializeValue(IEmitter emitter, Object value, Type type)
   at k8s.Yaml.SaveToString[T](T value)
   at K8s.JobSpinner.Service.Controllers.JobsController.WriteYaml(V1Job job) in C:\CarryCase\Venkatesh\M2c\KubernetesJobSDK\K8sJobSpinner\src\K8s.JobSpinner.Service\Controllers\JobsController.cs:line 76
   at K8s.JobSpinner.Service.Controllers.JobsController.CreateJobFromExisting(ReqCreateFromJob request) in C:\CarryCase\Venkatesh\M2c\KubernetesJobSDK\K8sJobSpinner\src\K8s.JobSpinner.Service\Controllers\JobsController.cs:line 63
   at lambda_method(Closure , Object )
   at Microsoft.Extensions.Internal.ObjectMethodExecutorAwaitable.Awaiter.GetResult()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at K8s.JobSpinner.Service.Middleware.SecurityMiddleware.Invoke(HttpContext context) in C:\CarryCase\Venkatesh\M2c\KubernetesJobSDK\K8sJobSpinner\src\K8s.JobSpinner.Service\Middleware\SecurityMiddleware.cs:line 36
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
 

Please help

**Comment by tg123** [comment]
hmmm 
could you please dump it with JSON format and i can take a look why it cause yaml failure


PS, json is a subset of yaml. you can use json as yaml

**Comment by VenkateshSrini** [comment]
@tg123 ,
Please find the JSON attached along with this email. Also could you please also tell me why I'm getting 'UnProcessibleEntity entity' error also when trying to clone an existing job.

`{"apiVersion":"batch/v1","kind":"Job","metadata":{"annotations":null,"clusterName":null,"creationTimestamp":"2021-01-25T09:52:48Z","deletionGracePeriodSeconds":null,"deletionTimestamp":null,"finalizers":null,"generateName":null,"generation":null,"labels":{"controller-uid":"2fcd20a3-1aaa-4e0f-a9b2-112e9cc409c0","job-name":"newcron"},"managedFields":[{"apiVersion":"batch/v1","fieldsType":"FieldsV1","fieldsV1":{"f:spec":{"f:backoffLimit":{},"f:completions":{},"f:parallelism":{},"f:template":{"f:spec":{"f:containers":{"k:{\"name\":\"samplecron\"}":{".":{},"f:args":{},"f:image":{},"f:imagePullPolicy":{},"f:name":{},"f:resources":{},"f:terminationMessagePath":{},"f:terminationMessagePolicy":{}}},"f:dnsPolicy":{},"f:restartPolicy":{},"f:schedulerName":{},"f:securityContext":{},"f:terminationGracePeriodSeconds":{}}}}},"manager":"unknown","operation":"Update","time":"2021-01-25T09:52:48Z"},{"apiVersion":"batch/v1","fieldsType":"FieldsV1","fieldsV1":{"f:status":{"f:completionTime":{},"f:conditions":{".":{},"k:{\"type\":\"Complete\"}":{".":{},"f:lastProbeTime":{},"f:lastTransitionTime":{},"f:status":{},"f:type":{}}},"f:startTime":{},"f:succeeded":{}}},"manager":"kube-controller-manager","operation":"Update","time":"2021-01-25T09:52:53Z"}],"name":"newcron1","namespace":"default","ownerReferences":null,"resourceVersion":"949","selfLink":"/apis/batch/v1/namespaces/default/jobs/newcron","uid":"2fcd20a3-1aaa-4e0f-a9b2-112e9cc409c0"},"spec":{"activeDeadlineSeconds":null,"backoffLimit":6,"completions":1,"manualSelector":null,"parallelism":1,"selector":{"matchExpressions":null,"matchLabels":{"controller-uid":"2fcd20a3-1aaa-4e0f-a9b2-112e9cc409c0"}},"template":{"metadata":{"annotations":null,"clusterName":null,"creationTimestamp":null,"deletionGracePeriodSeconds":null,"deletionTimestamp":null,"finalizers":null,"generateName":null,"generation":null,"labels":{"controller-uid":"2fcd20a3-1aaa-4e0f-a9b2-112e9cc409c0","job-name":"newcron"},"managedFields":null,"name":null,"namespace":null,"ownerReferences":null,"resourceVersion":null,"selfLink":null,"uid":null},"spec":{"activeDeadlineSeconds":null,"affinity":null,"automountServiceAccountToken":null,"containers":[{"args":["cloned"],"command":null,"env":null,"envFrom":null,"image":"venkateshsrini3/croncontr","imagePullPolicy":"Always","lifecycle":null,"livenessProbe":null,"name":"samplecron","ports":null,"readinessProbe":null,"resources":{"limits":null,"requests":null},"securityContext":null,"startupProbe":null,"stdin":null,"stdinOnce":null,"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","tty":null,"volumeDevices":null,"volumeMounts":null,"workingDir":null}],"dnsConfig":null,"dnsPolicy":"ClusterFirst","enableServiceLinks":null,"ephemeralContainers":null,"hostAliases":null,"hostIPC":null,"hostNetwork":null,"hostPID":null,"hostname":null,"imagePullSecrets":null,"initContainers":null,"nodeName":null,"nodeSelector":null,"overhead":null,"preemptionPolicy":null,"priority":null,"priorityClassName":null,"readinessGates":null,"restartPolicy":"OnFailure","runtimeClassName":null,"schedulerName":"default-scheduler","securityContext":{"fsGroup":null,"fsGroupChangePolicy":null,"runAsGroup":null,"runAsNonRoot":null,"runAsUser":null,"seLinuxOptions":null,"supplementalGroups":null,"sysctls":null,"windowsOptions":null},"serviceAccount":null,"serviceAccountName":null,"shareProcessNamespace":null,"subdomain":null,"terminationGracePeriodSeconds":30,"tolerations":null,"topologySpreadConstraints":null,"volumes":null}},"ttlSecondsAfterFinished":null},"status":null}`

**Comment by VenkateshSrini** [comment]
@tg123 ,
any findings that you could make based on the information that I shared ?

**Comment by tg123** [comment]
hi 

I load the v1pod obj from your json and test with code below
cannot repo

```
            var pod = JsonConvert.DeserializeObject<V1Pod>(File.ReadAllText(@"D:\tmp\x\json.txt"));
            Console.WriteLine(Yaml.SaveToString(pod));
```

output

```
k8s.Models.V1Pod
apiVersion: batch/v1
kind: Job
metadata:
  creationTimestamp: 2021-01-25T09:52:48.0000000Z
  labels:
    controller-uid: 2fcd20a3-1aaa-4e0f-a9b2-112e9cc409c0
    job-name: newcron
  managedFields:
  - apiVersion: batch/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:spec:
        f:backoffLimit: {}
        f:completions: {}
        f:parallelism: {}
        f:template:
          f:spec:
            f:containers:
              k:{"name":"samplecron"}:
                .: {}
                f:args: {}
                f:image: {}
                f:imagePullPolicy: {}
                f:name: {}
                f:resources: {}
                f:terminationMessagePath: {}
                f:terminationMessagePolicy: {}
            f:dnsPolicy: {}
            f:restartPolicy: {}
            f:schedulerName: {}
            f:securityContext: {}
            f:terminationGracePeriodSeconds: {}
    manager: unknown
    operation: Update
    time: 2021-01-25T09:52:48.0000000Z
  - apiVersion: batch/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:status:
        f:completionTime: {}
        f:conditions:
          .: {}
          k:{"type":"Complete"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:status: {}
            f:type: {}
        f:startTime: {}
        f:succeeded: {}
    manager: kube-controller-manager
    operation: Update
    time: 2021-01-25T09:52:53.0000000Z
  name: newcron1
  namespace: default
  resourceVersion: "949"
  selfLink: /apis/batch/v1/namespaces/default/jobs/newcron
  uid: 2fcd20a3-1aaa-4e0f-a9b2-112e9cc409c0
spec: {}
```

**Comment by VenkateshSrini** [comment]
@tg123 ,

I again tried. I'm getting the below exception only 

> System.Reflection.TargetException: Object does not match target type.
>    at System.Reflection.RuntimeMethodInfo.CheckConsistency(Object target)
>    at System.Reflection.RuntimeMethodInfo.InvokeArgumentsCheck(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
>    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
>    at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
>    at YamlDotNet.Helpers.GenericDictionaryToNonGenericAdapter.DictionaryEnumerator.get_Key()
>    at YamlDotNet.Helpers.GenericDictionaryToNonGenericAdapter.DictionaryEnumerator.get_Entry()
>    at YamlDotNet.Helpers.GenericDictionaryToNonGenericAdapter.DictionaryEnumerator.get_Current()
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseDictionary[TContext](IObjectDescriptor dictionary, IObjectGraphVisitor`1 visitor, Type keyType, Type valueType, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseObject[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.Traverse[TContext](Object name, IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseProperties[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseObject[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.Traverse[TContext](Object name, IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseList[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseObject[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.Traverse[TContext](Object name, IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseProperties[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseObject[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.Traverse[TContext](Object name, IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseProperties[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.TraverseObject[TContext](IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.Traverse[TContext](Object name, IObjectDescriptor value, IObjectGraphVisitor`1 visitor, TContext context, Stack`1 path)
>    at YamlDotNet.Serialization.ObjectGraphTraversalStrategies.FullObjectGraphTraversalStrategy.YamlDotNet.Serialization.IObjectGraphTraversalStrategy.Traverse[TContext](IObjectDescriptor graph, IObjectGraphVisitor`1 visitor, TContext context)
>    at YamlDotNet.Serialization.SerializerBuilder.ValueSerializer.SerializeValue(IEmitter emitter, Object value, Type type)
>    at k8s.Yaml.SaveToString[T](T value)
>    at K8s.JobSpinner.Service.Controllers.JobsController.WriteYaml(V1Job job) in C:\CarryCase\Venkatesh\M2c\KubernetesJobSDK\K8sJobSpinner\src\K8s.JobSpinner.Service\Controllers\JobsController.cs:line 79
>    at K8s.JobSpinner.Service.Controllers.JobsController.CreateJobFromExisting(ReqCreateFromJob request) in C:\CarryCase\Venkatesh\M2c\KubernetesJobSDK\K8sJobSpinner\src\K8s.JobSpinner.Service\Controllers\JobsController.cs:line 66
>    at lambda_method(Closure , Object )
>    at Microsoft.Extensions.Internal.ObjectMethodExecutorAwaitable.Awaiter.GetResult()
>    at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
>    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
>    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
>    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
>    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
>    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
>    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
>    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
>    at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
>    at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
>    at K8s.JobSpinner.Service.Middleware.SecurityMiddleware.Invoke(HttpContext context) in C:\CarryCase\Venkatesh\M2c\KubernetesJobSDK\K8sJobSpinner\src\K8s.JobSpinner.Service\Middleware\SecurityMiddleware.cs:line 36
>    at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
> 
> HEADERS
> =======
> Connection: keep-alive
> Content-Type: application/json
> Accept: text/plain
> Accept-Encoding: gzip, deflate, br
> Accept-Language: en-US,en;q=0.9
> Host: localhost:8080
> Referer: http://localhost:8080/swagger/index.html
> User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36
> Origin: http://localhost:8080
> Content-Length: 137
> API-Key: 66e26fb11c924df887d5f43b421927b2
> Sec-Fetch-Site: same-origin
> Sec-Fetch-Mode: cors
> Sec-Fetch-Dest: empty
I'm using the  below code 
```
var k8sClient = _k8sFactory.K8sClient;
            V1Container container;
            var existingJob = await k8sClient.ReadNamespacedJobAsync(request?.ExistigJobName, request.NameSpace);
            if (!string.IsNullOrEmpty(request?.ContainerName))
                container = existingJob.Spec.Template.Spec.Containers?
                            .FirstOrDefault(cntair => cntair.Name.CompareTo(request?.ContainerName) == 0);
            else
                container = existingJob.Spec.Template.Spec.Containers[0];

            if (request.Commands.Any())
                container.Command = request.Commands;
            if (request.JobParameters.Any())
                container.Args = request.JobParameters;
            
            var newJob = new V1Job
            {
                ApiVersion = existingJob.ApiVersion,
                Kind = existingJob.Kind,
                Metadata = existingJob.Metadata,
                Spec = existingJob.Spec
            };
            newJob.Metadata.Name = request?.NewJobName;
            newJob.Metadata.NamespaceProperty = request?.NameSpace;
            
            await WriteAsJson(newJob);
            await WriteYaml(newJob);
            var newJobCreationResult = await k8sClient.CreateNamespacedJobAsync(newJob, request?.NameSpace);
            var response = new ResCreateFromCronJob
            {
                JobName = newJobCreationResult.Name(),
                JobUid = newJobCreationResult.Uid(),
                NameSpace = newJobCreationResult.Namespace()
            };
            _logger.LogInformation("New job created scuccessfully");
            return Ok(response);
private async Task WriteYaml(V1Job job)
        {
            var yamlString = Yaml.SaveToString<V1Job>(job);
            await System.IO.File.WriteAllTextAsync("newjob.yaml", yamlString);
        }
        private async Task WriteAsJson(V1Job job)
        {
            var jsonString = JsonConvert.SerializeObject(job);
            await System.IO.File.WriteAllTextAsync("newjob.json",jsonString);
        }
```
The json is getting generated but not YAML. Attached is the JSON string for the same. I'm using the .NET core 3.1 and Kubernetes SDK 2.0.30.
[newjob.txt](https://github.com/kubernetes-client/csharp/files/5884466/newjob.txt)

**Comment by tg123** [comment]
Hi
after carefully investigate your issue I found it was a bug in 2.0 with old YAML library
fixed in 1f16a7f147ad2d00e7b9e90b00f51d12d7f1e801


please update to 4.0

**Comment by VenkateshSrini** [comment]
@tg123 ,

Thanks, it worked
